AppCenterAnalytics/AppCenterAnalytics/MSAnalytics.m

```diff
@@ -374,6 +382,27 @@ - (void)sendLog:(id<MSLog>)log flags:(MSFlags)flags {
   [self.channelUnit enqueueItem:log flags:flags];
 }
 
+- (void)setTransmissionInterval:(NSUInteger)interval {
+  if (interval > 24 * 60 * 60 || interval < 3) {
+    MSLogWarning([MSAnalytics logTag],
+                 @"Interval property is in invalid period, it should be greater than 2 second and lower than 1 day's seconds.");
+    return;
+  }
+
+  if (self.started) {
+    MSLogWarning([MSAnalytics logTag], @"Set transmission interval API must be called before MSAnalytics service starts.");
+    return;
+  }
+
+  self.flushInterval = interval;
+  self.channelUnitConfiguration = [[MSChannelUnitConfiguration alloc] initWithGroupId:[self groupId] flushInterval:self.flushInterval];
```

Comment:
Please do not create new `channelUnitConfiguration` here, just use the `flushInterval ` setter to change the value.